File: src/main/java/ro/nicolaemariusghergu/queryit/dto/ProductDto.java
package ro.nicolaemariusghergu.queryit.dto;

import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@EqualsAndHashCode
public class ProductDto {

    private Long id;

    private String name;

    private Double price;

    private Integer quantity;

    private String iconUrl;

    private CategoryDto category;

    private PromotionDto promotion;
}

File: src/main/kotlin/ro/nicolaemariusghergu/queryit/dto/ProductDto.kt
package ro.nicolaemariusghergu.queryit.dto

import java.math.BigDecimal

data class ProductDto (
    val id: Long,
    val name: String,
    val price: BigDecimal,
    val quantity: Int,
    val iconPath: String,
    val category: CategoryDto,
    val promotion: PromotionDto
)